### PR TITLE
C3: detect metered Android hotspot

### LINKED ISSUE #23731: Improve metered connection settings for prime lite
Better detection of phone hotspots:
- [x] C3 Android (https://github.com/commaai/openpilot/pull/23734)
- [ ] C3 iOS
- [ ] C3, allow forcing metered property from wifi settings menu on a connection basis.

Manual override
- [ ] Mark wifi as metered
- [x] Mark cellular connection as unmetered

### PR DESCRIPTION
**Pull request by pd0wm**
Resolves #23731

### CODE REVIEW

**Comment by adeebshihadeh** [comment]
I was thinking we make this is a separate property. For example, I'd like to treat my hotspot as normal wifi since it's unlimited. I imagine prime lite users with unlimited sims feel similarly.

**Comment by pd0wm** [comment]
I did it like this since a lot things currently depend on just the networkType (uploader, connect, internal scripts). We can also write a metered property with the wifi networkmanager config manually, where you can force mark it as unmetered.

File: selfdrive/hardware/tici/hardware.py [hunk 1]
```diff
@@ -91,11 +99,10 @@ def get_network_type(self):
       primary_connection = self.nm.Get(NM, 'PrimaryConnection', dbus_interface=DBUS_PROPS, timeout=TIMEOUT)
       primary_connection = self.bus.get_object(NM, primary_connection)
       primary_type = primary_connection.Get(NM_CON_ACT, 'Type', dbus_interface=DBUS_PROPS, timeout=TIMEOUT)
-      primary_id = primary_connection.Get(NM_CON_ACT, 'Id', dbus_interface=DBUS_PROPS, timeout=TIMEOUT)
 
       if primary_type == '802-3-ethernet':
         return NetworkType.ethernet
-      elif primary_type == '802-11-wireless' and primary_id != 'Hotspot':
```
Comment (pd0wm) [inline selfdrive/hardware/tici/hardware.py:98]: primary connection switches over properly when hotspot enabled, no need for this check.

**Comment by pd0wm** [comment]
Should be using the properties from networkmanager, which can be set on a connection by connection basis for both wifi and lte. We just need to build out the ui side to actually set them.

File: selfdrive/hardware/base.py [hunk 2]
```diff
@@ -67,6 +71,12 @@ def get_sim_info(self):
   def get_network_strength(self, network_type):
     pass
 
+  def get_network_metered(self, network_type) -> bool:
+    if network_type in [NetworkType.none, NetworkType.wifi, NetworkType.ethernet]:
```
Comment (adeebshihadeh) [inline selfdrive/hardware/base.py:75]:
```suggestion
    return network_type not in (NetworkType.none, NetworkType.wifi, NetworkType.ethernet)
```

Comment (pd0wm) [reply selfdrive/hardware/base.py:75]: Ah, of course. There used to be other code here, but that's now gone.

**Pull request merged**